ZFSin/zfs/module/zfs/zfs_windows_zvol_scsi.c

```diff
@@ -65,45 +65,162 @@
 /*
  * We have a list of ZVOLs, and we receive incoming (Target, Lun) requests that needs to be mapped
  * to the correct "zv" ptr.
+ *
+ * ssv-18807: fixed the race condition in the zvol destroy processing by adding remove lock logic
+ * to ensure no new I/O can be processed from the front end (StorPort) and all outstanding host I/Os 
+ * have left the pipeline.
+ *
+ * The zv control block starts to get protected in wzvol_assign_targetid() and this until wzvol_clear_targetid()
+ * is called.
```

Comment:
"and this until" ?

Reply:
Upon return from the call to wzvol_clear_targetid() we are then 100% sure that no new front-end I/O can enter the processing pipeline.  Which makes it safe to then free the zv based on existing protections from the back-end processing logic.